extension/data.ts

```diff
@@ -296,93 +287,308 @@ class RcxDict {
+    ベ: 'べ',
+    ペ: 'ぺ',
+    ホ: 'ほ',
+    ボ: 'ぼ',
+    ポ: 'ぽ',
+    マ: 'ま',
+    ミ: 'み',
+    ム: 'む',
+    メ: 'め',
+    モ: 'も',
+    ャ: 'ゃ',
+    ヤ: 'や',
+    ュ: 'ゅ',
+    ユ: 'ゆ',
+    ョ: 'ょ',
+    ヨ: 'よ',
+    ラ: 'ら',
+    リ: 'り',
+    ル: 'る',
+    レ: 'れ',
+    ロ: 'ろ',
+    ヮ: 'ゎ',
+    ワ: 'わ',
+    ヲ: 'を',
+    ン: 'ん',
+    ヴ: 'ゔ',
+    ｧ: 'ぁ',
+    ｱ: 'あ',
+    ｨ: 'ぃ',
+    ｲ: 'い',
+    ｩ: 'ぅ',
+    ｳ: 'う',
+    ｪ: 'ぇ',
+    ｴ: 'え',
+    ｫ: 'ぉ',
+    ｵ: 'お',
+    ｶ: 'か',
+    ｶﾞ: 'が',
+    ｷ: 'き',
+    ｷﾞ: 'ぎ',
+    ｸ: 'く',
+    ｸﾞ: 'ぐ',
+    ｹ: 'け',
+    ｹﾞ: 'げ',
+    ｺ: 'こ',
+    ｺﾞ: 'ご',
+    ｻ: 'さ',
+    ｻﾞ: 'ざ',
+    ｼ: 'し',
+    ｼﾞ: 'じ',
+    ｽ: 'す',
+    ｽﾞ: 'ず',
+    ｾ: 'せ',
+    ｾﾞ: 'ぜ',
+    ｿ: 'そ',
+    ｿﾞ: 'ぞ',
+    ﾀ: 'た',
+    ﾀﾞ: 'だ',
+    ﾁ: 'ち',
+    ﾁﾞ: 'ぢ',
+    ｯ: 'っ',
+    ﾂ: 'つ',
+    ﾂﾞ: 'づ',
+    ﾃ: 'て',
+    ﾃﾞ: 'で',
+    ﾄ: 'と',
+    ﾄﾞ: 'ど',
+    ﾅ: 'な',
+    ﾆ: 'に',
+    ﾇ: 'ぬ',
+    ﾈ: 'ね',
+    ﾉ: 'の',
+    ﾊ: 'は',
+    ﾊﾞ: 'ば',
+    ﾊﾟ: 'ぱ',
+    ﾋ: 'ひ',
+    ﾋﾞ: 'び',
+    ﾋﾟ: 'ぴ',
+    ﾌ: 'ふ',
+    ﾌﾞ: 'ぶ',
+    ﾌﾟ: 'ぷ',
+    ﾍ: 'へ',
+    ﾍﾞ: 'べ',
+    ﾍﾟ: 'ぺ',
+    ﾎ: 'ほ',
+    ﾎﾞ: 'ぼ',
+    ﾎﾟ: 'ぽ',
+    ﾏ: 'ま',
+    ﾐ: 'み',
+    ﾑ: 'む',
+    ﾒ: 'め',
+    ﾓ: 'も',
+    ｬ: 'ゃ',
+    ﾔ: 'や',
+    ｭ: 'ゅ',
+    ﾕ: 'ゆ',
+    ｮ: 'ょ',
+    ﾖ: 'よ',
+    ﾗ: 'ら',
+    ﾘ: 'り',
+    ﾙ: 'る',
+    ﾚ: 'れ',
+    ﾛ: 'ろ',
+    ﾜ: 'わ',
+    ｦ: 'を',
+    ﾝ: 'ん',
+    ｳﾞ: 'ゔ',
+    '‌': '‌', // the non-joiner used in google docs
+    voicedSoundMark: 'ﾟ',
+    semiVoicedSoundMark: 'ﾞ',
+  };
+
+  isKana(charCode: number): boolean {
+    return (
+      (charCode >= 12352 && charCode <= 12447) ||
+      (charCode >= 12448 && charCode <= 12543) ||
+      (charCode >= 65382 && charCode <= 65437)
+    );
+  }
 
-  cv: number[] = [
-    0x30f4, 0xff74, 0xff75, 0x304c, 0x304e, 0x3050, 0x3052, 0x3054, 0x3056,
-    0x3058, 0x305a, 0x305c, 0x305e, 0x3060, 0x3062, 0x3065, 0x3067, 0x3069,
-    0xff85, 0xff86, 0xff87, 0xff88, 0xff89, 0x3070, 0x3073, 0x3076, 0x3079,
-    0x307c,
-  ];
-  cs: number[] = [0x3071, 0x3074, 0x3077, 0x307a, 0x307d];
+  convertToHiragana(str: string): string {
+    let result = '';
+    for (let i = 0; i < str.length; i++) {
+      const char = str.charAt(i);
+      const nextChar = i + 1 <= str.length - 1 ? str.charAt(i + 1) : null;
+      const semiVoiced =
+        nextChar === this.hiraganaLookup['semiVoicedSoundMark'];
+      const voiced = nextChar === this.hiraganaLookup['voicedSoundMark'];
+      const key = semiVoiced
+        ? char + nextChar
+        : voiced
+        ? char + nextChar
+        : char;
+      const hiragana = this.hiraganaLookup[key];
+      result += hiragana !== undefined ? hiragana : char;
+
+      if (semiVoiced || voiced) {
+        i++;
+      }
+    }
+    return result;
+  }
 
   wordSearch(
     word: string,
     doNames: boolean,
     max?: number
   ): DictEntryData | null {
     let i;
-    let u;
-    let v;
-    let reason: string;
-    let p;
     const trueLen = [0];
```

Comment:
It doesn't like you update this anymore, is that okay?

Reply:
Are we referring to trueLen? It is updated on line 713.

Reply:
That's the reading of `trueLen` though. I'm talking about the updating of it to be something other than `[0]`.

Is 
```
// need to keep real length because of the half-width semi/voiced
// conversion
trueLen[reason.length] = i + 1;
```

not needed anymore or are we representing it a new way?

Reply:
I think this bug should be fixed now. The refactor changes removed the need for `trueLen`. After that, I updated the `maxLen` to what the actual selection should be an local tests seems to be working as expected. 🙏